src/libcore/iter.rs

```diff
@@ -280,6 +289,16 @@ pub trait Iterator {
     /// different sized integer, the `zip` function provides similar
     /// functionality.
     ///
+    /// # Undefined overflow
+    ///
+    /// The method does no guarding against overflows, so enumerating more than
+    /// `usize::MAX` elements is undefined.
```

Comment:
The intention of the RFC (whether or not it is clearly stated is a different matter) was to have overflows wrap when debug assertions are off. In particular, from the section @alexcrichton links,

> The error conditions that can arise, and their defined results, are as follows. The intention is that the defined results are the same as the defined results today. The only change is that now a panic may result.